apps/aecontract/src/aect_state_tree.erl

```diff
@@ -149,18 +149,67 @@ root_hash(#contract_tree{contracts = CtTree}) ->
                      {'ok', binary(), aec_poi:poi()}
                    | {'error', 'not_present' | 'wrong_root_hash'}.
 add_poi(Id, #contract_tree{contracts = CtTree}, Poi) ->
-    aec_poi:add_poi(Id, CtTree, Poi).
+    case aec_poi:add_poi(Id, CtTree, Poi) of
+        {ok, Contract, ContractPoi} ->
+            add_store_to_poi(
+              aect_contracts:deserialize(Id, Contract),
+              CtTree,
+              ContractPoi);
```

Comment:
I see this `ContractPoi` is the same variable that would have been returned earlier (as third element of the tuple). Ok.

Reply:
(with added items for store)